yoti-sdk-api/src/main/java/com/yoti/api/client/identity/extension/ThirdPartyAttributeExtensionBuilder.java

```diff
@@ -0,0 +1,58 @@
+package com.yoti.api.client.identity.extension;
+
+import static com.yoti.api.client.spi.remote.util.Validation.notNull;
+import static com.yoti.api.client.spi.remote.util.Validation.notNullOrEmpty;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import com.yoti.api.client.AttributeDefinition;
+
+public class ThirdPartyAttributeExtensionBuilder implements ExtensionBuilder<ThirdPartyAttributeContent> {
+
+    public static final String TYPE = "THIRD_PARTY_ATTRIBUTE";
+
+    private Date expiryDate;
+    private List<AttributeDefinition> definitions;
+
+    public ThirdPartyAttributeExtensionBuilder() {
+        this.definitions = new ArrayList<>();
+    }
+
+    public ThirdPartyAttributeExtensionBuilder withExpiryDate(Date expiryDate) {
+        notNull(expiryDate, Property.EXPIRY_DATE);
+
+        this.expiryDate = new Date(expiryDate.getTime());
+        return this;
+    }
+
+    public ThirdPartyAttributeExtensionBuilder withDefinition(String definition) {
+        notNullOrEmpty(definition, Property.DEFINITION);
+
+        this.definitions.add(new AttributeDefinition(definition));
+        return this;
+    }
+
+    public ThirdPartyAttributeExtensionBuilder withDefinitions(List<String> definitions) {
+        List<AttributeDefinition> attributeDefinitions = new ArrayList<>();
+        for (String definition : definitions) {
+            attributeDefinitions.add(new AttributeDefinition(definition));
+        }
+        this.definitions = attributeDefinitions;
+        return this;
+    }
+
+    public Extension<ThirdPartyAttributeContent> build() {
+        ThirdPartyAttributeContent thirdPartyAttributeContent = new ThirdPartyAttributeContent(expiryDate, definitions);
+        return new Extension<>(TYPE, thirdPartyAttributeContent);
+    }
+
+    private static final class Property {
+
+        private static final String EXPIRY_DATE = "expiryDate";
```

Comment:
Just to double check, is the use of camelCase here intentional? in `ThirdPartyAttributeContent` it uses `expiry_date` so just checking if this is correct.

Reply:
All good! This builder will create an instance of `ThirdPartyAttributeContent` which defines correctly the properties as

```
    private static final class Property {

        private static final String DEFINITIONS = "definitions";
        private static final String EXPIRY_DATE = "expiry_date";

    }
```

Reply:
Maybe I didn't explain correctly, I know that `ThirdPartyAttributeContent` has `expiry_date`, what I'm asking (assuming that is correct) is why _this_ class (`ThirdPartyAttributeExtensionBuilder`) has `expiryDate` instead of `expiry_date`.... I guess these are not JSON properties but _fields_ and are used only for exception messages?

Even so, why the mismatch if they both refer to the same thing? or do they refer to different things?

Reply:
gonna change to match the JSON property. As you said this field is used for the exception msg